lib/TPP/Conversion/ConvertLinalgToXsmm/ConvertLinalgToXsmm.cpp

```diff
@@ -601,7 +615,8 @@ static FailureOr<BrgemmInfo> isMappableToBrgemm(linalg::LinalgOp linalgOp) {
   unsigned n = contractionDims->n[0];
   unsigned k = contractionDims->k.back();
   std::optional<unsigned> batch;
-  if (contractionDims->k.size() == 2)
+  unsigned extraVnniDim = vnni::utils::isInVnniLayout(linalgOp);
```

Comment:
This works for bf16 VNNI but will not work for bf8. Perhaps use `getVNNIBlockingFactor`?

Reply:
It doesn't check the dimension size but the number of reduction dimensions which is type independent.
Refactored to be more readable.